setup.py

```diff
@@ -107,7 +107,17 @@ def run(self):
         import build_sphinx
         build_sphinx.rebuild()
 
-if 'clean' not in sys.argv:
+# determine if this run requires building of Qt GUI ui->py
+build_qt = True
+# _standard_ commands which shouldn't trigger the Qt build
+no_build_commands = ['clean', 'sdist', 'bdist', 'check', 'upload', 'register', '--help']
+
+for c in no_build_commands:
+    if any([c in b for b in sys.argv]):
+        build_qt = False
+        break
```

Comment:
Do you need substring matching for your checks?

With a whitelist and no substring matching this could be simplified to:
```
build_qt = any(c in sys.argv for c in build_commands)
```

If you do need substring checks, probably better to force them to the beginning of the string:
```
is_build_command = lambda arg: any(arg.startswith(cmd) for cmd in build_commands)
build_qt = any(is_build_command(arg) for arg in sys.argv)
```
or even switch to re checks if you need more control.

Reply:
I don't need substrings. The only whitelisted commands will be easily spelled out.

Reply:
On reflection the whitelist question may be moot: if you do `python setup.py clean build` then setup.py will call convertUI then process the clean and build. If clean removes the UI then this is a bug, but I don't think we care that much.